### PR TITLE
update to es6 based hapi style guide and lab 7 + Dependencies update

### PR DESCRIPTION
**Pull request by vdeturckheim**
Related to #41 

## Depedencies

* Boom 2 -> 3
* Hoek 2 -> 3
* Items 1 -> 2
* Joi 5 || 6 -> 5 || 6 || 7
* Code 1 -> 2
* Lab 5 -> 7
* Catbox-memory 1 -> 2

## Style

Build is passing with Lab 7.x.x and Eslint activated.

## Engine

Node < 4.0 will not be supported anymore.

-------------------------------

It is my first Hapi contribution, I hope I did not make any mistake :innocent:

### CODE REVIEW

**Comment by Marsup** [comment]
Welcome! :) All dependencies should start with the ES6 versions then, if you can run glue you can run others too.

**Comment by Marsup** [comment]
What are all those spaces around parentheses?

**Comment by vdeturckheim** [comment]
@Marsup 
Thanks ^^
* For the spaces, sorry, Webstorm configuration, I'm fixing this.
* So what should I do for the dependencies ?

**Comment by vdeturckheim** [comment]
Sorry for the spamming -_-

File: lib/index.js [hunk 1]
```diff
@@ -78,17 +79,17 @@ exports.compose = function (manifest /*, [options], callback */) {
...
-        var plugins = [];
-        var parsed;
+        let plugins = [];
+        let parsed;
 
         if (Array.isArray(manifest.plugins)) {
-            for (var i = 0, l = manifest.plugins.length; i < l; i++) {
-                var pluginObject = manifest.plugins[i];
-                var keys = Object.keys(pluginObject);
+            for (let i = 0, l = manifest.plugins.length; i < l; i++) {
```
Comment (Marsup) [inline lib/index.js]: No need to cache the length anymore.

File: lib/index.js [hunk 2]
```diff
@@ -129,16 +130,16 @@ internals.parseServer = function (server, relativeTo) {
...
         server = Hoek.clone(server);
 
-        var caches = [];
-        var config = [].concat(server.cache);
+        const caches = [];
+        const config = [].concat(server.cache);
 
-        for (var i = 0, il = config.length; i < il; ++i) {
-            var item = config[i];
+        for (let i = 0, il = config.length; i < il; ++i) {
```
Comment (Marsup) [inline lib/index.js]: No need to cache the length anymore.

File: package.json [hunk 3]
```diff
@@ -15,19 +15,19 @@
...
   "dependencies": {
-    "boom": "2.x.x",
+    "boom": "3.x.x",
     "hapi": "8.x.x || 9.x.x || 10.x.x || 11.x.x",
-    "hoek": "2.x.x",
-    "items": "1.x.x",
-    "joi": "5.x.x || 6.x.x"
+    "hoek": "3.x.x",
+    "items": "2.x.x",
+    "joi": "5.x.x || 6.x.x || 7.x.x"
```
Comment (Marsup) [inline package.json]: 7 should be enough.

File: package.json [hunk 4]
```diff
@@ -15,19 +15,19 @@
     "hapi"
   ],
   "engines": {
-    "node": ">=0.10.32"
+    "node": ">=4.0"
   },
   "dependencies": {
-    "boom": "2.x.x",
+    "boom": "3.x.x",
     "hapi": "8.x.x || 9.x.x || 10.x.x || 11.x.x",
```
Comment (Marsup) [inline package.json]: 11 only seems like a reasonable choice.

File: test/index.js [hunk 5]
```diff
@@ -271,214 +272,215 @@ describe('compose()', function () {
...
             server: {
                 cache: './catbox-memory'
             }
         };
 
-        expect(function () {
+        expect(() => {
 
-            Glue.compose(manifest, { relativeTo: __dirname + '/badpath' }, function () { });
+            Glue.compose(manifest, { relativeTo: __dirname + '/badpath' }, () => ({ }));
```
Comment (Marsup) [inline test/index.js]: That's not equivalent, you're returning an empty object here.

File: test/index.js [hunk 6]
```diff
@@ -271,214 +272,215 @@ describe('compose()', function () {
...
             plugins: {
                 './helloworld.js': null
             }
         };
 
-        expect(function () {
+        expect(() => {
 
-            Glue.compose(manifest, { relativeTo: __dirname + '/badpath' }, function () { });
+            Glue.compose(manifest, { relativeTo: __dirname + '/badpath' }, () => ({ }));
```
Comment (Marsup) [inline test/index.js]: That's not equivalent, you're returning an empty object here.

File: test/index.js [hunk 7]
```diff
@@ -271,214 +272,215 @@ describe('compose()', function () {
...
+    it('throws on options not an object', (done) => {
 
-        var manifest = {};
+        const manifest = {};
 
-        expect(function () {
+        expect(() => {
 
-            Glue.compose(manifest, 'hello', function () { });
+            Glue.compose(manifest, 'hello', () => ({ }));
```
Comment (Marsup) [inline test/index.js]: That's not equivalent, you're returning an empty object here.

File: test/index.js [hunk 8]
```diff
@@ -271,214 +272,215 @@ describe('compose()', function () {
...
-        var manifest = {
+        const manifest = {
             connections: 'hello'
         };
 
-        expect(function () {
+        expect(() => {
 
-            Glue.compose(manifest, function () { });
+            Glue.compose(manifest, () => ({ }));
```
Comment (Marsup) [inline test/index.js]: That's not equivalent, you're returning an empty object here.

File: test/index.js [hunk 9]
```diff
@@ -271,214 +272,215 @@ describe('compose()', function () {
...
-        var manifest = {
+        const manifest = {
             connections: []
         };
 
-        expect(function () {
+        expect(() => {
 
-            Glue.compose(manifest, function () { });
+            Glue.compose(manifest, () => ({ }));
```
Comment (Marsup) [inline test/index.js]: That's not equivalent, you're returning an empty object here.

**Comment by Marsup** [comment]
Sorry for the spam as well, I haven't marked all the mistakes on the fat arrow in the tests but you get the idea.

**Comment by vdeturckheim** [comment]
Thanks for this feedback !

**Comment by Marsup** [comment]
LGTM, maybe needs a squash now. @csrl ?

**Comment by vdeturckheim** [comment]
Actually, build is broken since [GLUE#484](https://github.com/hapijs/lab/pull/484). I will push a new commit tonight (hopefully).

**Comment by vdeturckheim** [comment]
Build is passing again.

**Comment by csrl** [comment]
Thanks! I'll review and pull it in over the next few days.  Then once Issue #31 is resolved, we can push out Glue 3.0

**Comment by csrl** [comment]
I've pulled this commit in to the v3 branch.  It'll get brought along when that is merged to master.

**Pull request closed without merging**